File: src/main/java/org/nia/model/QuestItem.java
package org.nia.model;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.query.Query;
import org.nia.db.HibernateConfig;
import org.nia.logic.lists.items.EQuestItem;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

/**
 * @author Иван, 06.04.2017.
 */
@Entity
@Getter
@Setter
@Table(name = "cwt_QuestItem")
public class QuestItem extends AbstractEntity {
    @Id
    @Column()
    @GeneratedValue
    private Integer publicID;
    @ManyToOne
    @JoinColumn(name = "questID", nullable = false)
    private Quest quest;
    @Enumerated(EnumType.STRING)
    private EQuestItem questItem;
    @Column
    private int itemCount;

    public static List<QuestItem> getAll(Quest quest) {
        List<QuestItem> res = new ArrayList<>();
        SessionFactory factory = HibernateConfig.getSessionFactory();
        try (Session session = factory.openSession()) {
            Query<QuestItem> query = session.createQuery("FROM QuestItem WHERE quest.publicID = " + quest.getPublicID(), QuestItem.class);
            res = query.list();
        } catch (Exception ex) {
            ex.printStackTrace();
        }
        return res;
    }

    public boolean delete() {
        boolean res = false;
        SessionFactory factory = HibernateConfig.getSessionFactory();
        try (Session session = factory.openSession()) {
            Transaction tx = session.beginTransaction();
            session.delete(this);
            tx.commit();
            res = true;
        } catch (Exception ex) {
            ex.printStackTrace();
        }
        return res;
    }
}

File: src/main/java/ru/nia/tavern/model/QuestItem.kt
package ru.nia.tavern.model

import ru.nia.tavern.model.types.items.EQuestItem
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table

/**
 * @author Иван, 06.04.2017.
 */
@Entity
@Table(name = "cwt_QuestItem")
class QuestItem {
    @Id
    @Column
    @GeneratedValue
    var publicID: Int? = null
    @ManyToOne
    @JoinColumn(name = "questID", nullable = false)
    lateinit var quest: Quest
    @Enumerated(EnumType.STRING)
    var questItem: EQuestItem? = null
    @Column
    var itemCount: Int = 0

//    fun delete(): Boolean {
//        var res = false
//        val factory = HibernateConfig.getSessionFactory()
//        try {
//            factory.openSession().use { session ->
//                val tx = session.beginTransaction()
//                session.delete(this)
//                tx.commit()
//                res = true
//            }
//        } catch (ex: Exception) {
//            ex.printStackTrace()
//        }
//
//        return res
//    }
//
//    companion object {
//
//        fun getAll(quest: Quest): List<QuestItem> {
//            var res: List<QuestItem> = ArrayList()
//            val factory = HibernateConfig.getSessionFactory()
//            try {
//                factory.openSession().use { session ->
//                    val query = session.createQuery("FROM QuestItem WHERE quest.publicID = " + quest.publicID!!, QuestItem::class.java)
//                    res = query.list()
//                }
//            } catch (ex: Exception) {
//                ex.printStackTrace()
//            }
//
//            return res
//        }
//    }
}
